packages/application-components/src/components/main-pages/custom-form-main-page/custom-form-main-page.tsx

```diff
@@ -70,31 +64,52 @@ const CustomFormMainPage = (props: CustomFormMainPageProps) => {
 
   return (
     <PageWrapper>
-      <CustomViewsSelector
-        customViewLocatorCode={props.customViewLocatorCode}
-      />
       <MainPageContainer>
-        <Spacings.Stack scale="l">
-          {props.customTitleRow || (
-            <PageHeaderTitle
-              title={props.title ?? ''}
-              subtitle={props.subtitle}
-              titleSize="big"
-            />
-          )}
-          {!props.hideControls && props.formControls && (
+        {props.customTitleRow || (
+          <PageHeaderTitle
+            title={props.title ?? ''}
+            subtitle={props.subtitle}
+            titleSize="big"
+          />
+        )}
+        <CustomViewsSelector
+          margin={`${uiKitDesignTokens.spacing40} 0 0 0`}
+          customViewLocatorCode={props.customViewLocatorCode}
+        />
+        {!props.hideControls && props.formControls && (
+          <div
+            css={css`
+              margin-top: ${uiKitDesignTokens.spacing40};
+            `}
+          >
             <Spacings.Inline justifyContent="flex-end">
               {props.formControls}
             </Spacings.Inline>
-          )}
-          {!props.hideDivider && <Divider />}
-        </Spacings.Stack>
+          </div>
+        )}
+        {!props.hideDivider && (
+          <div
+            css={css`
+              margin-top: ${uiKitDesignTokens.spacing40};
+            `}
+          >
+            <Divider />
+          </div>
```

Comment:
Are all these custom styles necessary? Can't we just wrap everything with `<Spacings.Stack>` so that all siblings have the same spacing between them? 🤔

Reply:
I had to remove the `Stack` component as it was introducing extra margins when the `CustomViewsSelector` does not render anything (no Custom views configured for the provided locator).